cmd/platform-starter/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/inconshreveable/log15"
+	prompt "github.com/segmentio/go-prompt"
+
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "platform-starter"
+	app.Usage = "Initialize platform projects with common configuration."
+	app.Version = "1.0.0"
+	app.Action = run
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "dir",
+			Usage: "directory to initialize",
+			Value: ".",
+		},
+		cli.BoolFlag{
+			Name:  "npm",
+			Usage: "forces the usage of npm for installing dependencies",
+		},
+	}
+
+	app.Run(os.Args)
+}
+
+var requirements = []string{
+	"csscomb",
+	"editorconfig-tools",
+}
+
+type file struct {
+	asset    *asset
+	dest     []string
+	fromRoot bool
+}
+
+func (f file) path(root, wd string) string {
+	if f.fromRoot {
+		return filepath.Join(append([]string{root}, f.dest...)...)
+	}
+
+	return filepath.Join(append([]string{wd}, f.dest...)...)
+}
+
+var files = []file{
+	{mustAsset(configCsscombJson()), mkPath(".csscomb.json"), false},
+	{mustAsset(configEditorconfig()), mkPath(".editorconfig"), true},
+	{mustAsset(hooksPreCommit()), mkPath(".git", "hooks", "pre-commit"), true},
+}
+
+func run(ctx *cli.Context) error {
+	log15.Info("Starting platform-starter")
+
+	log15.Info("Installing requirements...")
+	for _, r := range requirements {
+		ensureInstalled(r, ctx.Bool("npm"))
+	}
+
+	root, err := os.Getwd()
+	if err != nil {
+		log15.Crit("unable to get current working directory", "err", err)
+		os.Exit(1)
+	}
+
+	if !isDir(".git") {
+		log15.Warn("Current directory is not a git repository.")
+		log15.Info("Initializing git repository...")
+		if err := cmd("git", "init"); err != nil {
+			log15.Crit("unable to initialize git repository", "err", err)
+			os.Exit(1)
+		}
+	}
+
+	dir := ctx.String("dir")
+	dir, err = filepath.Abs(dir)
```

Comment:
This is aimed at those project containing both the frontend and the backend. Typically you don't need the `--dir` because they're the same. In the case of having both this allows installing certain things like the `.csscomb.json` only inside the frontend part.